conda_smithy/configure_feedstock.py

```diff
@@ -567,13 +567,29 @@ def _collapse_subpackage_variants(
     # on osx, merge MACOSX_DEPLOYMENT_TARGET & c_stdlib_version to max of either; see #1884
     all_variants = _merge_deployment_target(all_variants, has_macdt)
 
-    top_level_loop_vars = list_of_metas[0].get_used_loop_vars(
-        force_top_level=True
-    )
-    top_level_vars = list_of_metas[0].get_used_vars(force_top_level=True)
+    # we use the intersection of all of the loop vars in the outputs
+    # to distribute CI jobs.
+    # this eliminates cases where a fully expanded build matrix would produce
+    # duplicate outputs if an output used some variables but not others.
+    # For outputs that do not use some variables, conda-build will properly
+    # deduplicate the packages it builds in a single CI job.
+    top_level_loop_vars = [
+        meta.get_used_loop_vars(force_top_level=True) for meta in list_of_metas
+    ]
+    top_level_vars = [
+        meta.get_used_vars(force_top_level=True) for meta in list_of_metas
+    ]
+    top_level_vars = set.intersection(*top_level_vars)
+    top_level_loop_vars = set.intersection(*top_level_loop_vars)
+
     if "target_platform" in all_used_vars:
         top_level_loop_vars.add("target_platform")
 
+    logger.info(
+        "variables used to distribute packages over CI jobs:\n%s",
+        textwrap.indent(pprint.pformat(top_level_loop_vars), "    "),
+    )
```

Comment:
FWIW do see the following when testing with the `python-feedstock`

```python
INFO:conda_smithy.configure_feedstock:variables used to distribute packages over CI jobs:
    {'build_type', 'channel_targets', 'target_platform'}
```

Note the presence of `channel_targets`

Reply:
Right I am just not sure if this means the README will be stable, which was the question.

Reply:
Posted more info about testing for that issue in comment: https://github.com/conda-forge/conda-smithy/issues/2296#issuecomment-2816417890